pkg/types/powervs/validation/machinepool.go

```diff
@@ -30,34 +31,37 @@ func ValidateMachinePool(p *powervs.MachinePool, fldPath *field.Path) field.Erro
 	}
 
 	// Validate Memory
-	if p.Memory != "" {
-		memory, err := strconv.ParseInt(p.Memory, 10, 64)
-		if err == nil {
-			if memory < 2 || memory > 64 {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "memory must be an integer number of GB that is at least 2 and no more than 64"))
-			}
-		} else {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "memory must be an integer number of GB that is at least 2 and no more than 64"))
+	if p.MemoryGiB != 0 {
```

Comment:
should we return an error if memory is zero?

Reply:
No, If the memory is 0, the default webhook will add the default value as 32(https://github.com/openshift/machine-api-operator/pull/998/commits/e7ab47bc98b714d8188179b71c2fe26119285718)